### DISCUSSION TITLE
How to best support typing for dictionary

### DISCUSSION THREAD

**Discussion by xinghul** [MichalLytek/type-graphql#845] [Q&A]
**Is your feature request related to a problem? Please describe.**
Feature request/question. I have an `InputType` named `Filters`, right now I want to create another `InputType` called `FiltersByName`, the interface is like this:
```typescript
interface IFiltersByName = Record<string, Filters>;

// or 

interface IFiltersByName {
  [name: string]: Filters;
}
```

**Question**
How do I type the `FiltersByName` for type-graphql?
```typescript
@InputType()
class FiltersByName implements IFiltersByName {
  ...?
}
```
 I understand this might be a limitation of graphql scheme itself, but would appreciate it in case you know a better solution.

**Describe alternatives you've considered**
Right now I'm using `GraphQLJSONObject` for `FiltersByName`, which is not ideal:
```typescript
@Query(() => any)
public async getAllCount(
  @Arg('params', () => GraphQLJSONObject) params: IFiltersByName,
) {
...
}
```

**Comment by MichalLytek** [comment]
I always recomend such pattern:
```ts
@InputType()
class FilterValueInput {
  @Field()
  key: string;

  @Field()
  value: Filter;
}

// ...

@Query(() => Foo)
public async getAllCount(
  @Arg('params', () => [FilterValueInput]) params: [FilterValueInput],
) {
  // ...
}
```

So you can then send filter values as:
```graphql
query {
  getAllCount(params: [{ key: 'foo', value: BAR }, { key: 'bar', value: FOO }]) {
    # ...
  }
}
```
